File: 백준/Silver/18258. 큐 2/큐 2.cc
#include <bits/stdc++.h>

using namespace std;
int n, i;
string s;
queue<int> q;

int main() {
    ios::sync_with_stdio(0), cin.tie(0);
    cin>>n;
    while(n--){
        cin>>s;
        if(s=="push") {
            int num;
            cin>>num;
            q.push(num);
        }
        if(s=="pop"){
            if(!q.empty()){
                cout<<q.front()<<"\n";
                q.pop();
            }
            else cout<<-1<<"\n";
        }
        if(s=="size"){
            cout<<q.size()<<"\n";
        }
        if(s=="empty"){
            if(q.empty()){
                cout<<1<<"\n";
            }
            else cout<<0<<"\n";
        }
        if(s=="front"){
            if(!q.empty()) cout<<q.front()<<"\n";
            else cout<<-1<<"\n";
        }
        if(s=="back"){
            if(!q.empty()) cout<<q.back()<<"\n";
            else cout<<-1<<"\n";
        }
    }
    return 0;
}

File: 백준/Silver/18258. 큐 2/큐 2.py
import sys
from collections import deque
r = sys.stdin.readline
sys.setrecursionlimit(10**6)

def solution():
    Q = deque()

    n = int(r().rstrip())
    for _ in range(n):
        ol = r().split()
        if ol[0] == 'push':
            Q.append(ol[1])
        elif ol[0]=='pop':
            if Q: print(Q.popleft())
            else: print(-1)
        elif ol[0]=='size':
            print(len(Q))
        elif ol[0]=='empty':
            if Q: print(0)
            else: print(1)
        elif ol[0]=='front':
            if Q: print(Q[0])
            else: print(-1)
        elif ol[0]=='back':
            if Q: print(Q[-1])
            else: print(-1)
                
solution()